### ISSUE TITLE
small ui suggestion for split views & new session buttons on headerbar

### ISSUE THREAD

**Issue by alex285** [gnunn1/tilix#934]
i use Adapta because the issue is a bit more obvious here, and also Adapta (and flat themes in general) is a quite popular theme

from left to right, excluding the appMenu we have
Session Buttons
New Session
Vertical Split
Horizontal Split

the two issues
1. i often press split instead of new, because buttons is super close and super similar
2. the session counting isn't visible enough

my suggestion which is a very light recommendation and i cant be sure without actually testing, is

1. to drop the split buttons from headerbar, because the right click popover inside terminal makes perfect sense (or S/C) and is much handier
2. to sub "+" with "New" ..which again im not sure what happens with translations strings
3. to somehow highlight more the session counting, perhaps with some coloring

![tilix-issue](https://cloud.githubusercontent.com/assets/11244308/26278308/ea85b51e-3d9f-11e7-9850-71b7d2f69abe.png)

**Comment by gnunn1** [comment]
Oops, committed that against the wrong issue.

**Comment by gnunn1** [comment]
Some thoughts on your suggestions:

1. I don't think dropping the split buttons from the headerbar is a good idea, this is the primary purpose of Tilix so I like having it front and center. Plus on a personal level I use them all the time and rarely bother with the context menu.

2. Again, I'm not keen on this as it is too wordy in my opinion. If someone wants to create an appropriate icon though instead of the + I'm not averse to using that.

3. Coloring is hard because it is theme specific and very difficult to pick a color that works across all themes.  Unfortunately themes only define a very limited number colors so there's not much options there.

I think a better way to handle this is to simply add some appropriate CSS styles that theme authors or uses such as yourself could use to modify the look and feel. Does that make sense to you?

**Comment by alex285** [comment]
@gnunn1 
>Plus on a personal level I use them all the time and rarely bother with the context menu

okay, then dropping them is not an option. so next step is how we can improve the current one. im attaching the same view with Adwaita theme

![tilix-adwaita](https://cloud.githubusercontent.com/assets/11244308/26281749/5ae4b706-3e09-11e7-87dd-56673b3c7a91.png)

here the sessions arrow is obvious that belongs to 1/1 button when in Adapta it seems like a separated button, since the distances with rest buttons are the same

again some suggestions
1. we could slightly modify the padding/margin
2. we could drop the caret arrow since it isn't super useful indicator plus it is repeated right after on title bar, and in Adapta is  confusing where it belongs. however this needs more ideas how can the session counter be more visible. use one of the API colors of GTK theme perhaps?
3. the plus "+" icon is not a very good icon for a "new" action, but not ideas for what icon it should be used. perhaps a terminal icon with a plus sign?
4. dont know if thats possible without affecting the height of headerbar, but all icons should be a bit bigger?

about the 4, in Adapta the buttons look like icon buttons, so it makes sense to be bigger, but in Adwaita they look like normal buttons that just have an icon inside 

also if you still planing to implement a tabs sidebar https://github.com/gnunn1/tilix/issues/681 (which btw it works perfect!!) this issue is gonna be obsolete!

**Comment by gnunn1** [comment]
To me though all of these items are issues with Adapta, isn't this really an issue for them to resolve? I'm willing to add some css classes if it helps them style things but I'm not changing  padding or icon sizes within tilix since from my perspective it looks fine on Adwaita.

I'm happy to consider changing the plus icon though if someone comes up with something nice.

**Comment by alex285** [comment]
@gnunn1 
> To me though all of these items are issues with Adapta

agree thus i open https://github.com/adapta-project/adapta-gtk-theme/issues/417

> I'm happy to consider changing the plus icon though

also agree

**Comment by gnunn1** [comment]
OK I'm closing this for now, if the Adapta folks request some changes in classes or whatever they can re-open it.

**Comment by tista500** [comment]
@gnunn1,

Hi, I'm an author of Adapta.
So is it possible to add 2 style-classes to that GtkToggleButton (placed at the left-side of 'Plus' image-button)?

 1. `.text-button` style-class
That GtkToggleButton has label(s) to show the current session ID and total number of sessions. If so, we need to optimize the base-line of label(s) and font-weight. Adwaita does not concern the base-lines and font-weights though... 

2. `.session-sidebar-button` style-class or something like that
I mean I want a widget specific style-class to tweak the spacing between label node and image (arrow) node inside the GtkBox container of GtkToggleButton. 

Regards.

**Comment by gnunn1** [comment]
@tista500 I have added the ```session-sidebar-button``` class, however I can't add the ```tex-button``` class because it widens the button excessively under Adwaita. See the screen shot below.

![screenshot from 2017-05-21 16-16-31](https://cloud.githubusercontent.com/assets/9358238/26287061/2df45ec6-3e41-11e7-85e6-d5a56f4c235e.png)

**Comment by tista500** [comment]
@gnunn1,

> I have added the session-sidebar-button class

Oh thanks!

> however I can't add the tex-button class because it widens the button excessively under Adwaita.

https://git.gnome.org/browse/gtk+/tree/gtk/theme/Adwaita/_common.scss#n569
But for me, your shot shows the natural size of `button.text-button` as we can see in many places when Adwaita was selected. If you're still keeping arrows inside that button, `button.toggle.text-button.image-button` might be the best in semantically, just my thought... At least, `.text-button` style-class brings theme's best (suggested) inner spacing and weight for label(s) in most places.

Anyway we can tweak the spacing without `.text-button` though.

Regards.

**Comment by gnunn1** [comment]
I saw you updated and closed the adapta issue, thanks for taking care of this @tista500.

**Comment by tista500** [comment]
Thanks for your work @gnunn1 :)

**Comment by bil-elmoussaoui** [comment]
Looks much better!
![screenshot from 2017-05-22 00-30-48](https://cloud.githubusercontent.com/assets/7660997/26287947/ec85955a-3e85-11e7-8919-12a9e7259a2b.png)
Thanks @gnunn1 @tista500 !
